pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -187,6 +261,280 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
+		if v1alphaTp.Spec.Extends == "" && len(newRules) == 0 && len(v1alphaTpCP.Spec.DisableRules) == 0 && !doContinue {
+			// flip the checkType of the TailoredProfile to the new checkType if the TailoredProfile does not extend any profile
+			// and it does not have any rules left after removing migrated rules and the before migration TailoredProfile has
+			// enableRules
+			if profileType == cmpv1alpha1.ScanTypeNode {
+				v1alphaTpCP.SetAnnotations(map[string]string{cmpv1alpha1.ProductTypeAnnotation: string(cmpv1alpha1.ScanTypePlatform)})
+			} else {
+				v1alphaTpCP.SetAnnotations(map[string]string{cmpv1alpha1.ProductTypeAnnotation: string(cmpv1alpha1.ScanTypeNode)})
+			}
+			v1alphaTpCP.Spec.EnableRules = v1alphaTp.Spec.EnableRules
+			logger.Info("Flipping the checkType of the TailoredProfile to the new checkType", "checkType", profileType)
+			r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileMigratedCheckType", "Flipping the checkType of the TailoredProfile to the new checkType: %s", profileType)
+		}
+	}
+
+	if v1alphaTp.Spec.Extends == "" && len(v1alphaTpCP.Spec.DisableRules) == 0 && len(v1alphaTpCP.Spec.EnableRules) == 0 {
+		errorMsg := "TailoredProfile does not have any rules left after removing migrated rules and it does not extend any profile"
+		v1alphaTpCP.Status.State = cmpv1alpha1.TailoredProfileStateError
+		v1alphaTpCP.Status.ErrorMessage = errorMsg
+		doContinue = false
+		logger.Info(errorMsg)
+		r.Eventf(v1alphaTp, corev1.EventTypeWarning, "TailoredProfileMigratedRule", errorMsg)
+	}
+
+	if !doContinue {
+		err = r.Client.Update(context.TODO(), v1alphaTpCP)
+		logger.Info("Updating TailoredProfile with migrated rules removed")
+		if err != nil {
+			return false, err
+		}
+	}
+	return doContinue, nil
+}
+
+func isValidationRequired(tp *cmpv1alpha1.TailoredProfile) bool {
+	if tp.Spec.Extends != "" {
+		if tp.Spec.DisableRules != nil || tp.Spec.EnableRules != nil || tp.Spec.ManualRules != nil || tp.Spec.SetValues != nil {
+			return true
+		}
+	} else {
+		if tp.Spec.EnableRules != nil || tp.Spec.ManualRules != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *ReconcileTailoredProfile) handleDeprecation(
+	v1alphaTp *cmpv1alpha1.TailoredProfile, logger logr.Logger) (bool, error) {
+
+	deprecatedRules, err := r.getDeprecatedRules(v1alphaTp, logger)
+	if err != nil {
+		return false, err
+	}
+
+	deprecatedVariables, err := r.getDeprecatedVariables(v1alphaTp, logger)
+	if err != nil {
+		return false, err
+	}
+
+	doContinue := true
+	v1alphaTpCP := v1alphaTp.DeepCopy()
+
+	// remove any deprecated variables from the tailored profile
+	if len(v1alphaTp.Spec.SetValues) > 0 && len(deprecatedVariables) > 0 {
+		var newVariables []cmpv1alpha1.VariableValueSpec
+		for vi := range v1alphaTp.Spec.SetValues {
+			variables := &v1alphaTp.Spec.SetValues[vi]
+			if _, ok := deprecatedVariables[variables.Name]; ok {
+				doContinue = false
+				logger.Info("Removing deprecated variable", "variable", variables.Name)
```

Comment:
Does this logger include the TailoredProfile name or should that be included in the logging string explicitly?

Reply:
the reconcile loop should contain the name of tailored profile, see line 95
```
	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```